docs/advisors/checks/postgresql-vacuum-sanity-check.md

```diff
@@ -0,0 +1,50 @@
+# PostgreSQL Vacuum sanity
+
+
+## Description
+
+This check performs a quick gathering of some basic vacuum parameters and alerts if some default values are found along with the critical alert for autovacuum not being enabled.
+
+## Resolution
+*   Autovacuum is critical for PostgreSQL to maintain table bloat under control along with transaction wraparound. If autovacuum is turned off, this could lead to serious table bloat, performance issues and an accelerated path to transaction wraparound. Typically, the autovacuum parameter is turned off only because there is a very specific and special reason for it and most likely another task may be vacuuming on a different schedule. If this is not the case, consider enabling this as soon as possible.
+*   If autovacuum\_max\_workers is set to the default value of 3 and you find that the vacuum is not cleaning up as expected, adjust this parameter to increase the number of processes to perform the autovacuum.
+*   The default autovacuum\_scale\_factor value is 20%. This means that a table will not get autovacuumed until 20% of the data in it has been updated or deleted + a threshold.  For small tables, this is not too much of a concern. However, on large tables this can be problematic. In order to mitigate this at the table level, identify large tables that are not being vacuumed frequently and apply custom storage parameters to the table.  For example,
+
+`ALTER TABLE foobar SET ( autovacuum\_vacuum\_scale\_factor = .1)`
+
+The above will set the scale factor to 10% on the foobar table.  
+
+Another option is to set the autovacuum to take place when _**n**_ number of rows have been updated or deleted.
+
+`ALTER TABLE foobar SET ( autovacuum\_vacuum\_scale\_factor = 0, autovacuum\_vacuum\_threshold = 100000 )`
+
+In the above example, the foobar table will autovacuum when 100,000 rows have been updated or deleted. 
+
+For vacuum cost settings, keep the following in mind.
+
+During the execution of VACUUM and ANALYZE commands, the system maintains an internal counter that keeps track of the estimated cost of the various I/O operations that are performed. When the accumulated cost reaches a limit specified in the vacuum\_cost\_limit parameter, the process performing the operation will sleep for a time, as specified by vacuum\_cost\_delay. Then it will reset the counter and continue execution.
```

Comment:
```suggestion
During the execution of VACUUM and ANALYZE commands, the system maintains an internal counter that keeps track of the estimated cost of the various I/O operations that are performed. When the accumulated cost reaches a limit specified in the **vacuum\_cost\_limit** parameter, the process performing the operation will sleep for a time, as specified by **vacuum\_cost\_delay**. Then it will reset the counter and continue execution.
```